src/main/java/org/jabref/gui/externalfiles/AutoLinkFilesAction.java

```diff
@@ -34,47 +33,62 @@
-        BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
-        List<BibEntry> entries = stateManager.getSelectedEntries();
-
-        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
-        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
+        final BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
+        final List<BibEntry> entries = stateManager.getSelectedEntries();
+        final AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
                 database,
                 preferences.getFilePreferences(),
                 preferences.getAutoLinkPreferences(),
                 ExternalFileTypes.getInstance());
-        Task<List<BibEntry>> linkFilesTask = new Task<>() {
+        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
+
+        if (entries.stream().mapToInt(entry -> entry.getFiles().size()).sum() == 0) {
+            dialogService.notify(Localization.lang("The selected entries do not have any file links attached."));
+            return;
+        }
 
+        Task<AutoSetFileLinksUtil.LinkFilesResult> linkFilesTask = new Task<>() {
             @Override
-            protected List<BibEntry> call() {
+            protected AutoSetFileLinksUtil.LinkFilesResult call() {
                 return util.linkAssociatedFiles(entries, nc);
             }
 
             @Override
             protected void succeeded() {
-                if (!getValue().isEmpty()) {
-                    if (nc.hasEdits()) {
-                        nc.end();
-                        undoManager.addEdit(nc);
-                    }
-                    dialogService.notify(Localization.lang("Finished automatically setting external links."));
-                } else {
-                    dialogService.notify(Localization.lang("Finished automatically setting external links.") + " " + Localization.lang("No files found."));
+                AutoSetFileLinksUtil.LinkFilesResult result = getValue();
+
+                if (!result.getFileExceptions().isEmpty()) {
+                    dialogService.notify(Localization.lang("Problem finding files. See error log for details."));
+                    return;
                 }
+
+                if (result.getBibEntries().isEmpty()) {
+                    dialogService.notify(Localization.lang("Finished automatically setting external links.")
+                            + " " + Localization.lang("No files found."));
+                    return;
+                }
+
+                if (nc.hasEdits()) {
+                    nc.end();
+                    undoManager.addEdit(nc);
+                }
+
+                dialogService.notify(Localization.lang(
+                        "Finished automatically setting external links. Changed %0 entries.",
```

Comment:
Reuse `Localization.lang("Finished automatically setting external links.")` from above by splitting the `Localization.lang(...)` here.